integration-tests/android-has-interface/src/test/java/com/example/ExampleFragmentInjectionActivityTest.java

```diff
@@ -0,0 +1,27 @@
+package com.example;
+
+import androidx.test.core.app.ActivityScenario;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static androidx.lifecycle.Lifecycle.State.CREATED;
+import static androidx.test.core.app.ActivityScenario.launch;
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(RobolectricTestRunner.class) //
+public class ExampleFragmentInjectionActivityTest {
+    @Test
+    public void activityFragmentInjection() {
+        try (ActivityScenario<ExampleFragmentInjectionActivity> scenario = launch(ExampleFragmentInjectionActivity.class)) {
+            scenario.moveToState(CREATED);
+            scenario.onActivity(activity -> {
+                ExampleFragment fragment = (ExampleFragment) activity.getFragmentManager().findFragmentByTag("ExampleFragment");
+                assertThat(fragment.string).isEqualTo("Hello!");
+                assertThat(fragment.aLong).isEqualTo(10L);
+                assertThat(fragment.anInt).isEqualTo(20);
+            });
+        }
+    }
+}
```

Comment:
This test is failing when the task `:integration-tests:android-has-interface:testReflectDebugUnitTest` is run. But as a single test run from the IDE, it passes just fine. Any hints what might be wrong? @JakeWharton

Reply:
It's probably because of moveToState. `launch` may have an optional parameter for initial state like [Fragment](https://github.com/TWiStErRob/repros/blob/master/espresso/fragment-lifecycle/src/androidTest/java/com/example/FragmentLifecycleTest.kt#L30),  even if it doesn't, the default initial state may be RESUMED. I think you are going to DESTROYED with this code. _Just an idea, not fully sure._

Also you may need to idle the looper, since Robolectric 4.4.

Reply:
Thanks for the pointers. I'm gonna investigate the Fragment/ActivityScenario apis more. I should add that it fails with the following error:

```
No injector factory bound for Class<com.example.ExampleFragment>
java.lang.IllegalArgumentException: No injector factory bound for Class<com.example.ExampleFragment>
	at dagger.android.DispatchingAndroidInjector.inject(DispatchingAndroidInjector.java:136)
	at dagger.android.support.AndroidSupportInjection.inject(AndroidSupportInjection.java:75)
	at dagger.android.support.AndroidSupportInjection.inject(AndroidSupportInjection.java:67)
	at com.example.ExampleFragment.onAttach(ExampleFragment.java:34)
```